torchrl/envs/common.py

```diff
@@ -848,42 +1142,58 @@ def step(self, tensordict: TensorDictBase) -> TensorDictBase:
             if dims
             else next_tensordict_out.shape
         )
-        expected_reward_shape = torch.Size(
-            [*leading_batch_size, *self.reward_spec.shape]
-        )
-        actual_reward_shape = reward.shape
-        if actual_reward_shape != expected_reward_shape:
-            reward = reward.view(expected_reward_shape)
-            next_tensordict_out.set(self.reward_key, reward)
+        for reward_key in self.reward_keys:
+            reward = next_tensordict_out.get(reward_key)
+            expected_reward_shape = torch.Size(
+                [
+                    *leading_batch_size,
+                    *self.output_spec["_reward_spec"][reward_key].shape,
+                ]
+            )
+            actual_reward_shape = reward.shape
+            if actual_reward_shape != expected_reward_shape:
+                reward = reward.view(expected_reward_shape)
+                next_tensordict_out.set(reward_key, reward)
 
         # TODO: Refactor this using done spec
-        done = next_tensordict_out.get(self.done_key)
-        # unsqueeze done if needed
-        expected_done_shape = torch.Size([*leading_batch_size, *self.done_spec.shape])
-        actual_done_shape = done.shape
-        if actual_done_shape != expected_done_shape:
-            done = done.view(expected_done_shape)
-            next_tensordict_out.set(self.done_key, done)
+        for done_key in self.done_keys:
+            done = next_tensordict_out.get(done_key)
+            expected_done_shape = torch.Size(
+                [
+                    *leading_batch_size,
+                    *self.output_spec["_done_spec"][done_key].shape,
```

Comment:
```suggestion
                    *self.output_spec["_done_spec", done_key].shape,
```
if it works with nested done_key :p